graphula-core/src/Graphula/Internal.hs

```diff
@@ -51,6 +51,11 @@ class GHasDependenciesRecursive fieldsProxy node deps where
   genericDependsOnRecursive :: fieldsProxy -> node -> deps -> node
 
 -- This instance head only matches EoT representations of
+-- datatypes with no constructors. Ignore dependencies.
+instance GHasDependencies (Proxy nodeTy) (Proxy depsTy) Void deps where
```

Comment:
Here's what I was thinking. Given:
```haskell
data Nope deriving Generic
```

a user might try to set `type Dependencies Nope` to one of three things:
1. `()`, implicitly or explicitly
2. `Void`, explicitly - this isn't completely unreasonable, but then the user can't use `node @Nope`. Instead they have to say `nodeWith @Nope (undefined :: Void)`, which is probably not what they mean. We actually probably want to disallow this for all instances.
3. Anything else - this is probably a mistake. 

I figured out how to handle all three cases, but it seems to require overlapping instances:
```haskell
-- Only allow empty data decls to use () as their dependencies
instance {-# OVERLAPPING #-} GHasDependencies (Proxy nodeTy) (Proxy depsTy) Void (Either () Void) where
   genericDependsOn _ _ node _ = node

-- Disallow any other kind of non-void, non-unit dependencies for empty data decls
instance
  {-# OVERLAPPABLE #-}
  ( TypeError
    ( 'Text "A datatype with no constructors can't use the dependencies in" ':$$:
      DependenciesTypeInstance nodeTy depsTy
    )
  ) => GHasDependencies (Proxy nodeTy) (Proxy depsTy) Void (Either deps rest) where
    genericDependsOn _ _ _ _ = error "Impossible"

-- Don't let the user specify `Void` as a dependency for any node
instance
  ( TypeError
    ( 'Text "Use ‘()’ instead of ‘Void’ for datatypes with no dependencies in" ':$$:
      DependenciesTypeInstance nodeTy depsTy
    )
  ) => GHasDependencies (Proxy nodeTy) (Proxy depsTy) node Void where
    genericDependsOn _ _ _ _ = error "Impossible"
```

I'm not actually sure why the last one doesn't seem to need overlapping instances.